app/core/edge_inference.py

```diff
@@ -111,24 +111,39 @@ class EdgeInferenceManager:
     INFERENCE_SERVER_URL = "inference-service:8000"
     MODEL_REPOSITORY = MODEL_REPOSITORY_PATH
 
-    def __init__(self, config: Dict[str, LocalInferenceConfig] | None, verbose: bool = False) -> None:
+    def __init__(
+        self,
+        inference_configs: Dict[str, LocalInferenceConfig] | None,
+        edge_config: Optional[RootEdgeConfig] = None,
```

Comment:
We have a bit of a weird set up right now where we use a config as a detector arguments in `inference_config`, but specific values for `always_return_cloud_prediction` and `disable_cloud_escalation`.

We want the new `min_time_between_escalations` field to be configurable on a detector level, and I think it makes sense to keep it with `always_return_edge_prediction` and `disable_cloud_escalation` since it's only valid with a particular configuration of those fields. However, the best place I saw to keep track of the time of last escalation for a detector is in `EdgeInferenceManager` -- hence adding the `edge_config` input. 

I think it might be better to get rid of the `local_inference_template` in the detector config entirely, and just add the fields from it to the detector config. Alternatively, we could move all three of the new detector config fields into the `local_inference_template`. That reorganization felt unrelated to this PR, but happy to do it if others agree it should be done.

Reply:
Definitely agree that either getting rid of `local_inference_template` or moving the current detector config fields into it would be good, but yeah makes sense to do that in a different PR.